build.project

```diff
@@ -30,21 +23,44 @@ if [ -z "$TARGET" ]; then
     exit 1
 fi
 
+
 TARGET_FAMILY=$(echo "$TARGET" | awk -F- '{ print $1 }')
-TARGET_SUBFAMILY="$(echo "$TARGET" | awk -F- '{ print $1"-"$2 }')"
+TARGET_SUBFAMILY="$(echo "$TARGET" | awk -F- '{ print $2 }')"
+TARGET_PROJECT="$(echo "$TARGET" | awk -F- '{ print $3 }')"
+
+
+BUILD_TARGET_FILE="_targets/build.project.${TARGET_FAMILY}-${TARGET_SUBFAMILY}"
+BUILD_PROJECT_FILE="_projects/build.project.${TARGET_FAMILY}-${TARGET_SUBFAMILY}-${TARGET_PROJECT}"
+
+
+if [ "$TARGET" != "$TARGET_FAMILY-$TARGET_SUBFAMILY" ] || [ "$TARGET" != "$TARGET_FAMILY-$TARGET_SUBFAMILY-$TARGET_PROJECT" ]; then    
```

Comment:
shit, my bad, I've done an error, You should put `&&` in here instead of `||`, otherwise it's always true

Reply:
Event putting `&&` for `$TARGET = armv7m7-imxrt106x-` it does not work

Reply:
yeah, You're right, we should check if `TARGET_PROJECT` is non-empty. I propose more hackish approach:
```bash
if [ "$TARGET" != "$TARGET_FAMILY-$TARGET_SUBFAMILY${TARGET_PROJECT:+-$TARGET_PROJECT}" ]; then
```
but I don't know if the readability decrease is worth doing it that way

Reply:
Your approach also works.